docs/advanced-topics/dockstore-cli/dockstore-cli-faq.rst

```diff
@@ -0,0 +1,95 @@
+to containers compared to what is available on your host machine. You
+can change what it allocates using the Docker for Mac GUI under
+``Preferences > Advanced`` as described
+`here <https://docs.docker.com/docker-for-mac/#advanced>`__.
+
+* The default allocation can cause workflows or tools to fail without informing the user with a memory or resource related error message. If you find that your workflow or tool is behaving differently on a Mac compared to a similarly resourced Ubuntu environment, you can try increasing the resources allocated to Docker on the Mac to resolve the discrepancy.
+
+How can I use the Dockstore CLI with Python 2?
+----------------------------------------------
+
+Python 2 support ended in 2020. You can get the Python 2 requirements.txt with ``curl -o requirements.txt "https://dockstore.org/api/metadata/runner_dependencies?python_version=2"``
+but it is currently untested.
+
+How do I launch tools/workflows without internet access on compute nodes?
+-------------------------------------------------------------------------
+
+Some tools/workflows require Docker images to launch even if they are
+local entries. If the compute nodes do not have internet access, you can
+follow these steps:
+
+1. download the Docker image(s) on the head node which does have internet access using the ``docker save -o <filename> <imagename>``
+2. ensure that the ``<imagename>`` matches the image name specific in the CWL or WDL descriptor 
+3. place the image file(s) in a location that the compute nodes have access to (make sure there are only images in that directory)
+4. specify in the dockstore config file (default ~/.dockstore/config) the directory that contains your image(s) using ``docker-images = /home/user/docker_images_directory``
+
+The Dockstore CLI will automatically load all Docker images in the
+directory specified prior to a ``launch --local-entry`` command.
+
+.. _cromwell-docker-lockup:
+
+I was running a WDL locally, but some of my tasks returned 137 and/or now I cannot use Docker.
+----------------------------------------------------------------------------------------------
+
+This is a known issue with Cromwell, which the Dockstore CLI uses to launch WDL workflows. It is due to how Cromwell manages resources on a local machine, which it must do differently compared to running on the cloud. As such, it is much more likely to happen if you are running a computationally intensive scattered task, such as LD pruning 23 chromosomes where each chromosome is an instance of a scattered task. In these scenarios, Cromwell might lockup the Docker service. If a Docker lockup happens, you will notice tasks do not progress beyond WaitingForReturnCode and you will be temporarily unable to "spin up" any Docker containers, even outside of Cromwell. Thankfully, this state can be resolved by restarting the Docker service via the Docker Desktop dropdown, or entering ``service docker restart`` on the command line.
```

Comment:
Should probably indicate which OS and OS version this behaviour was seen on. I don't have a Docker Desktop dropdown.  `service docker restart` probably wouldn't work without `sudo`

Reply:
I will rewrite the solutions to be more OS-specific, but the problem itself... this occurred to me on Mac OS but I see no reason why it would not occur on any other OS Cromwell can run on provided Cromwell is running in "local mode."

Reply:
Marc confirmed this can happen on Linux.